tensorboard/defs/web.bzl

```diff
@@ -377,3 +379,65 @@ tf_web_library = rule(
         "_closure_library_base": CLOSURE_LIBRARY_BASE_ATTR,
     }.items()),
     outputs=DEPRECATED_CLUTZ_OUTPUTS)
+
+def _tf_web_test_python_stub_impl(ctx):
+  ctx.actions.expand_template(
+      template=ctx.file._template,
+      output=ctx.outputs.main,
+      substitutions={
+          "{BINARY_PATH}": ctx.executable.web_library.short_path,
+          "{WEB_PATH}": ctx.attr.web_path,
+      }
+  )
+
+_tf_web_test_python_stub = rule(
+    implementation=_tf_web_test_python_stub_impl,
+    attrs={
+        "web_library": attr.label(executable=True, cfg="host", mandatory=True),
+        "web_path": attr.string(mandatory=True),
+        "main": attr.output(),
+        "_template": attr.label(
+            default=Label("//tensorboard/defs:web_test_python_stub.template.py"),
+            allow_single_file=True,
+        ),
+    },
+)
+
+def tf_web_test(name, web_library, src, **kwargs):
+  """Run tests defined by a `tf_web_library`.
```

Comment:
I added docs here and removed the docs on the internal-only stub rule
above. I think that this makes it easier to read the internal stub rule
and to identify that it’s mostly trivial, and it also reduces some
duplication now that we have docs on the public macro. If you disagree,
let me know and I’d be happy to add back the internal docs.